lib/pipefy_message/logging.rb

```diff
@@ -47,5 +47,17 @@ def self.logger_setup
     def self.included(base)
       base.extend(self)
     end
+
+    ##
+    # Adds context, correlation and event identifiers to logs. Should be
+    # called with the hash containing the remainder of the logs as
+    # argument, eg: logger.info(log_context({ ... }, context, correlation_id, event_id)).
+    def log_context(log_hash, context, correlation_id, event_id)
```

Comment:
**question:** Tbh, I'm not really happy with this solution; having to pass `context, correlation_id, event_id` as arguments each time is only marginally better than having to add that entire hash. I wish this could be automagically picked up from the calling scope without the caller having to do anything specific. Any ideas on a straightforward way to do that?

Reply:
We can take a look at Thread and Fiber Local variables

https://docs.ruby-lang.org/en/2.7.0/Thread.html#class-Thread-label-Fiber-local+vs.+Thread-local

Reply:
Are you suggesting `thread_variable_set`? I'm not sure, messing with global variables -- which is what it looks like that would result in, in practice -- seems like it could lead to too much trouble down the line just for simplifying some logs :sweat_smile: I thought maybe we could pass that method a [binding](https://ruby-doc.org/core-2.7.0/Binding.html) for the calling scope, at least we'd have a single context parameter rather than three ([or two, if we drop](https://github.com/pipefy/pipefy_message/pull/33/files#r934832526) `context`). Though we'd still have to assume the variables have the same name everywhere, I'm not sure that's good either...

Reply:
Yes, it can be dangerous but is a possibility. Anyway, we can see that later.